src/main/java/de/tum/in/www1/artemis/repository/CourseRepository.java

```diff
@@ -354,6 +360,32 @@ default Course findWithEagerLearningGoalsByIdElseThrow(long courseId) {
         return findWithEagerLearningGoalsById(courseId).orElseThrow(() -> new EntityNotFoundException("Course", courseId));
     }
 
+    /**
+     * Utility method to check if the messaging feature is enabled for a course.
+     * <p>
+     * Keep in sync with the implementation in {@link CourseInformationSharingConfiguration}
+     *
+     * @param courseId the id of the course
+     * @return true if the messaging feature is enabled for the course, false otherwise
+     */
+    default boolean isMessagingEnabled(long courseId) {
```

Comment:
Would it make sense to provide overloads that pass a `Course` object as parameter?
Then in cases where we have got the course loaded anyway there is no additional DB-call. Same for the method below.

Reply:
Done! Good idea